### PR TITLE
Adds a script to use debug version of MapboxNavigationNative

### PR DESCRIPTION
**Pull request by S2Ler**
This is MVP and by no means perfect solution. Possible improvements:
- [x] Automatic destination selection
- [ ] Support device builds
- [ ] Make these builds on NavNative CI instead of building locally

### CODE REVIEW

File: scripts/debug_navigation_native.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+set -x
+set -e
+
+# Install a tool which parses JSONs
+brew list jq || brew install jq
+
+NATIVE_PATH=$1
+DESTINATION=$2
+NATIVE_VERSION=$(jq '.object.pins[] | select(.package=="MapboxNavigationNative") | .state.version' ./Package.resolved | sed 's/^"\(.*\)"$/\1/')
```
Comment (bamx23) [inline scripts/debug_navigation_native.sh:18]: I guess you can inline a python code here and not install `jq`

File: scripts/debug_navigation_native.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+fi
+
+set -x
+set -e
+
+# Install a tool which parses JSONs
+brew list jq || brew install jq
+
+NATIVE_PATH=$1
+DESTINATION=$2
```
Comment (bamx23) [inline scripts/debug_navigation_native.sh:17]: Can we use "any simulator" here?

File: scripts/debug_navigation_native.sh (hunk 1 above)
Comment (S2Ler) [reply scripts/debug_navigation_native.sh]: thank you for suggestion, implemented

File: scripts/debug_navigation_native.sh (hunk 2 above)
Comment (S2Ler) [reply scripts/debug_navigation_native.sh]: "generic/platform=iOS Simulator" works. Thank you

File: scripts/debug_navigation_native.sh [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+NATIVE_PATH=$1
+DESTINATION="generic/platform=iOS Simulator"
+NATIVE_VERSION=$(python3 << END
+import json
+import sys
+
+packageResolved = json.loads(open("Package.resolved", "r").read())
+for pin in packageResolved["object"]["pins"]:
+    if pin["package"] == "MapboxNavigationNative":
+        sys.stdout.write(pin["state"]["version"])
```
Comment (bamx23) [inline scripts/debug_navigation_native.sh:22]: NIT: You can use `revision` here and get the commit hash

**Review by bamx23 — APPROVED** [review]
Great work, thanks!

**Pull request merged**